lib/shoulda/matchers/action_controller/route_params.rb

```diff
@@ -26,17 +28,24 @@ def controller_and_action_given_as_string?
         def extract_params_from_string
           controller, action = args[0].split('#')
           params = (args[1] || {}).merge(controller: controller, action: action)
-          stringify_values(params)
+          normalize_values(params)
         end
 
         def stringify_params
-          stringify_values(args[0])
+          normalize_values(args[0])
         end
 
-        def stringify_values(hash)
-          hash.inject({}) do |hash_copy, (key, value)|
-            hash_copy[key] = stringify(value)
-            hash_copy
+        def normalize_values(hash)
+          hash.each_with_object({}) do |(key, value), hash_copy|
+            hash_copy[key] = symbolize_or_stringify(key, value)
+          end
+        end
+
+        def symbolize_or_stringify(key, value)
+          if key.in?(VALUES_TO_SYMBOLIZE)
+            value.to_sym
```

Comment:
We need the `value.to_sym` if we want accept the format as string or symbol.

Since we accept controller and action as string and symbols, I think is a good idea we accept the `format` as string and symbols too. What you think?